From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Wed, 15 Apr 2026 23:09:00 +0200
Subject: appFavorites: Move installed-changed where it belongs

In the dash we are monitoring the installed applications and reload the
favorites, while this is not something that is related to the dash per
se, and if the AppFavorites is used to track the favorites elsewhere it
would not be updated when new favorite apps are installed.

So move the logic to AppFavorites, that will now emit a changed
event whenever any favorite changed.

Origin: https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/4180
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/gnome-shell/+bug/2148339
---
 js/ui/appFavorites.js | 7 +++++--
 js/ui/dash.js         | 6 ++----
 2 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/js/ui/appFavorites.js b/js/ui/appFavorites.js
index a88f98c..d4017b5 100644
--- a/js/ui/appFavorites.js
+++ b/js/ui/appFavorites.js
@@ -12,6 +12,10 @@ class AppFavorites extends Signals.EventEmitter {
         this._parentalControlsManager.connect('app-filter-changed', () =>
             this._updateFavorites());
 
+        this._appSystem = Shell.AppSystem.get_default();
+        this._appSystem.connect('installed-changed', () =>
+            this._updateFavorites());
+
         this.FAVORITE_APPS_KEY = 'favorite-apps';
         this._favorites = {};
         global.settings.connect(`changed::${this.FAVORITE_APPS_KEY}`, () =>
@@ -41,8 +45,7 @@ class AppFavorites extends Signals.EventEmitter {
 
     reload() {
         const ids = global.settings.get_strv(this.FAVORITE_APPS_KEY);
-        const appSys = Shell.AppSystem.get_default();
-        const apps = ids.map(id => appSys.lookup_app(id))
+        const apps = ids.map(id => this._appSystem.lookup_app(id))
                       .filter(app => app !== null && this._parentalControlsManager.shouldShowApp(app.app_info));
         this._favorites = {};
         for (let i = 0; i < apps.length; i++) {
diff --git a/js/ui/dash.js b/js/ui/dash.js
index 17967f9..17c1a25 100644
--- a/js/ui/dash.js
+++ b/js/ui/dash.js
@@ -378,10 +378,8 @@ export const Dash = GObject.registerClass({
 
         this._appSystem = Shell.AppSystem.get_default();
 
-        this._appSystem.connect('installed-changed', () => {
-            AppFavorites.getAppFavorites().reload();
-            this._queueRedisplay();
-        });
+        this._appSystem.connect('installed-changed', () =>
+            this._queueRedisplay());
         AppFavorites.getAppFavorites().connect('changed', this._queueRedisplay.bind(this));
         this._appSystem.connect('app-state-changed', this._queueRedisplay.bind(this));
 
